docs/styleguides/python.md

```diff
@@ -0,0 +1,173 @@
+# DCOS Python Style Guide
+###### Version 0.1
+###### April-6-2017
+
+For Python source code within DCOS, we prefer to keep Google Python Style Guide Revision 2.59 as the base and **extend** it with
+some changes that will make DCOS codebase look modern, consistent and easy to follow.
+
+##### Rationale
+
+With any style guide, the primary requirement is to adopt one and follow it consistently. With our choices below, we can use tools available in the ecosystem to our advantage, enforce consistency in the codebase, creating and publishing automatically generated documentation for our code.
+
+##### Versioned document
+
+Styleguide _conventions_ are not set in stone. We can improve upon this or remove inconvenient styles by creating a PR, discussing the merits or demerits to arrive at a consensus.
+
+#### Base Style Guide
+
+* [Google Python Style Guide Revision 2.59](https://google.github.io/styleguide/pyguide.html)
+
+#### Summary of Style Requirements DCOS which extend Base Style Guide and are specific to DCOS.
+
+* Import Requirements
+    * Adopt the base style guide preferences.
+
+        ```
+        Use import x for importing packages and modules.
+        Use from x import y where x is the package prefix and y is the module name with no prefix.
+        Use from x import y as z if two modules named y are to be imported or if y is an inconveniently long name.
+        ```
+    * Use `__init__.py` only to represent python packages.
+        * Avoid importing of objects in `__init__.py` for readability.
+* Import formatting
+
+    * The types of imports should be separated by single line and order is should be.
+        * standard library imports.
+        * third-party imports.
+        * application specific imports.
+
+    * The `from` style import and `import` statements should be separated by single line too.
+    * Each group of these imports should be sorted.
+
+For Example
+
+```
+import json
+import uuid
+
+from subprocess import CalledProcessError
+from typing import Callable, List
+
+import retrying
+
+from retrying import retry, RetryError
+from teamcity.messages import TeamcityServiceMessages
+
+import test_util.aws
+import test_util.cluster
+
+from pkgpanda.util import logger, write_string
+```
+
+* [Packages](https://google.github.io/styleguide/pyguide.html?showone=Packages#Packages)
+    * Import each module using the full pathname location of the module.
+    * No Relative Imports.
+
+* Maximum line length is 120 characters.
```

Comment:
Could you please provide some more context for that decision?

Also, at the very beginning, we state that we "extend" the google style guideline (https://github.com/dcos/dcos/pull/1439/files#diff-a301a44012e0a06325b181e5835c6ce1R5), yet the google style guideline clearly states:
```
Maximum line length is 80 characters.
```

Personally, I think that if your code extends beyond 80 columns then you are either:
* nesting too much and should think about refactoring your code into smaller functions
* your functions have too many arguments which is bad for testability anyway
* should break the line somehow

In my opinion, having hard limit of 90-100 (flake8 starts to complain) and "soft" limit (the one we try to follow) of 80 seems like a good compromise. 

We can have a hard limit of 120 so that we can handle/lint old code while encouraging people to adhere to 80 lines.